expr/func_constantline.go

```diff
@@ -0,0 +1,54 @@
+package expr
+
+import (
+	"fmt"
+
+	"github.com/grafana/metrictank/api/models"
+	"github.com/grafana/metrictank/schema"
+)
+
+type FuncConstantLine struct {
+	value float64
+	from  uint32
+	to    uint32
+}
+
+func NewConstantLine() GraphiteFunc {
+	return &FuncConstantLine{}
+}
+
+func (s *FuncConstantLine) Signature() ([]Arg, []Arg) {
+	return []Arg{
+			ArgFloat{key: "value", val: &s.value},
+		}, []Arg{
+			ArgSeriesList{},
+		}
+}
+
+func (s *FuncConstantLine) Context(context Context) Context {
+	s.from = context.from - 1
+	s.to = context.to - 1
+	return context
+}
+
+func (s *FuncConstantLine) Exec(dataMap DataMap) ([]models.Series, error) {
+	out := pointSlicePool.Get().([]schema.Point)
+	out = append(out,
+		schema.Point{Val: s.value, Ts: s.from},
+		schema.Point{Val: s.value, Ts: s.from + uint32((s.to-s.from)/2.0)},
+		schema.Point{Val: s.value, Ts: s.to},
+	)
```

Comment:
it is guaranteed at this point (due to prior validation) that from < to.
however, from may equal to -1, which would result in duplicate timestamps. this is an edge case, and causes graphite to crash, but we may as well handle it :)